### PR TITLE
GRL - instancing support

### PR DESCRIPTION
**Pull request by RolandCsibrei**

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14460/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14460/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14460/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14460/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14460/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14460/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14460/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14460/merge#BCU1XR#0

File: packages/dev/core/src/Shaders/greasedLine.vertex.fx [hunk 1]
```diff
@@ -39,7 +42,8 @@ void main() {
         vec3 grlNext = grl_nextAndCounters.xyz;
         grlCounters = grl_nextAndCounters.w;
 
-        mat4 grlMatrix = worldViewProjection;
+        mat4 grlMatrix = worldViewProjection * finalWorld;
```
Comment (Popov72) [inline packages/dev/core/src/Shaders/greasedLine.vertex.fx:45]: It should be `viewProjection * finalWorld` instead.

**Review by Popov72 — CHANGES_REQUESTED** [review]

File: packages/dev/core/src/Shaders/greasedLine.vertex.fx (hunk 1 above)
Comment (RolandCsibrei) [reply packages/dev/core/src/Shaders/greasedLine.vertex.fx]: thank you, fixed

**Review by Popov72 — APPROVED** [review]

File: packages/dev/core/src/Shaders/greasedLine.vertex.fx (hunk 1 above)
Comment (RolandCsibrei) [reply packages/dev/core/src/Shaders/greasedLine.vertex.fx]: sry, one more thing to fix

Comment (RolandCsibrei) [reply packages/dev/core/src/Shaders/greasedLine.vertex.fx]: @Popov72 
Is this correct? Can I drop these three uniorms?
```javascript
                uniforms: [
                    // "worldViewProjection",
                    // "worldView",
                    // "view",
                    "world",
                    "viewProjection",
                    "projection",
```

**Comment by RolandCsibrei** [comment]
@Popov72 
Is this correct? Can I drop these three uniorms?
```javascript
                uniforms: [
                    // "worldViewProjection",
                    // "worldView",
                    // "view",
                    "world",
                    "viewProjection",
                    "projection",
```

File: packages/dev/core/src/Shaders/greasedLine.vertex.fx (hunk 1 above)
Comment (Popov72) [reply packages/dev/core/src/Shaders/greasedLine.vertex.fx]: Yes, if you don't use them yourself you can remove them.

**Comment by Popov72** [comment]
Yes (see my answer in the comments).

File: packages/dev/core/src/Shaders/greasedLine.vertex.fx (hunk 1 above)
Comment (RolandCsibrei) [reply packages/dev/core/src/Shaders/greasedLine.vertex.fx]: OIk, thank you!

**Comment by bjsplat** [comment]
Visualization tests for webgl1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14460/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

**Comment by bjsplat** [comment]
Visualization tests for webgl2 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14460/merge/testResults/webgl2/index.html

If tests were successful afterwards, this report might not be available anymore.

**Comment by bjsplat** [comment]
Visualization tests for webgl1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14460/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

**Comment by bjsplat** [comment]
Visualization tests for webgl1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14460/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

**Review by sebavan — APPROVED** [review]

**Pull request merged**